### PR TITLE
simplified device_query calls for sw_emu,hw_emu

### PR DESCRIPTION
**Pull request by vboggara-xilinx**
simplified device_query calls for sw_emu, hw_emu

### CODE REVIEW

File: src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.cxx [hunk 1]
```diff
@@ -1889,22 +1889,20 @@ int CpuemShim::xclCloseContext(const uuid_t xclbinId, unsigned int ipIndex) cons
...
   if (xclemulation::config::getInstance()->getIsPlatformEnabled()) {
-    return (mQueryTable["m2m"] == "enabled" ? true : false);
-  }
-  return false;
-}
-
-bool CpuemShim::isNoDMAEnabled() {
-  if (xclemulation::config::getInstance()->getIsPlatformEnabled()) {
-    return (mQueryTable["nodma"] == "enabled" ? true : false);
+    if (queryKey == key_type::nodma) {
```
Comment (keryell) [inline src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.cxx:1900]: While you are here, can you remove the previous code obfuscation on all these `return`?

File: src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.h [hunk 2]
```diff
@@ -246,7 +247,7 @@ namespace xclcpuemhal2 {
       uint64_t mReqCounter;
       FeatureRomHeader mFeatureRom;
       boost::property_tree::ptree mPlatformData;
-      std::map<std::string, std::string> mQueryTable;
+      std::map<key_type, std::string> mQueryTable;
```
Comment (keryell) [inline src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.h:250]: At the end I wonder whether an array or a `std::vector<std::string>` indexed by an equivalent of an `enum` is not what we want here...

File: src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/device_hwemu.cxx [hunk 3]
```diff
@@ -31,31 +31,17 @@ namespace {
 
   static std::map<query::key_type, std::unique_ptr<query::request>> query_tbl;
 
-  struct m2m
+  struct deviceQuery
   {
-    using result_type = query::m2m::result_type;
+    using result_type = uint32_t;
```
Comment (keryell) [inline src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/device_hwemu.cxx:36]: Why not a `bool` everywhere?

File: src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.cxx (hunk 1 above)
Comment (vboggara-xilinx) [reply src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.cxx]: comments addressed

File: src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/device_hwemu.cxx (hunk 3 above)
Comment (vboggara-xilinx) [reply src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/device_hwemu.cxx]: comments addressed

File: src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.h (hunk 2 above)
Comment (vboggara-xilinx) [reply src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.h:250]: std::vector<std::string> not required in this case as we dont need to maintain the vector with max size of enum.

**Comment by vboggara-xilinx** [comment]
can someone merge these changes?

File: src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.h [hunk 4]
```diff
@@ -263,7 +263,7 @@ using addr_type = uint64_t;
 
       std::string getSimulatorType(const std::string& binaryDirectory);
       void createPreSimScript(const std::string& wcfgFilePath, std::string& preSimScriptPath);
-      void constructQueryTable();
+      void constructQueryTable();
```
Comment (keryell) [inline src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.h:266]: Do not add useless space.
Please check your code before committing and configure your text-editor or IDE.

File: src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx [hunk 5]
```diff
@@ -1935,24 +1935,17 @@ uint32_t HwEmShim::getAddressSpace (uint32_t topology)
...
-
-  bool HwEmShim::isNoDMAEnabled() {
-    if (xclemulation::config::getInstance()->getIsPlatformEnabled()) {
-      return (mQueryTable["nodma"] == "enabled" ? true : false);
-    }
-    return false;
+  // New API's for device query - m2m and no-dma
+  int HwEmShim::deviceQuery(key_type queryKey) {
+    if (mQueryTable.find(queryKey) != mQueryTable.end())
+      return (mQueryTable[queryKey] == "enabled" ? 1 : 0);
```
Comment (keryell) [inline src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx:1947]: I am not convinced by this code.

**Pull request merged**